0.preprocess-sites/4.image-and-segmentation-qc.py

```diff
@@ -248,90 +268,105 @@
-    + gg.geom_point(gg.aes(fill=corr_qc_col), shape="s", size=6)
-    + gg.geom_text(gg.aes(label="site_location"), color="lightgrey", size=6)
-    + gg.facet_wrap("~Metadata_Well")
-    + gg.coord_fixed()
-    + gg.theme_bw()
-    + gg.ggtitle(
-        f"Correlation between BC & CP DAPI images\n(Alignment quality)\n{plate}"
-    )
-    + gg.theme(
-        axis_text=gg.element_blank(),
-        axis_title=gg.element_blank(),
-        strip_background=gg.element_rect(colour="black", fill="#fdfff4"),
-    )
-    + gg.scale_fill_cmap(name="Correlation")
-)
-
-output_file = pathlib.Path(
-    output_figuresdir, f"plate_layout_BC_to_CP_DAPI_correlation_per_well.png",
-)
-if check_if_write(output_file, force, throw_warning=True):
-    correlation_gg.save(output_file, dpi=300, verbose=False)
-
-# Plot final compartment thresholds per well
-threshold_col_prefix = "Threshold_FinalThreshold_"
-for threshhold_compartment in ["Cells", "Nuclei"]:
-    threshold_full_col = f"{threshold_col_prefix}{threshhold_compartment}"
-
-    if threshold_full_col not in image_df.columns:
-        continue
-
-    image_df_subset = image_df.dropna(subset=[threshold_full_col])
-
-    compartment_finalthresh_gg = (
-        gg.ggplot(image_df_subset, gg.aes(x="x_loc", y="y_loc"))
-        + gg.geom_point(gg.aes(fill=f"{threshold_full_col}"), shape="s", size=6)
+for plate in platelist:
+    correlation_gg = (
+        gg.ggplot(
+            image_df_subset.loc[image_df_subset["Metadata_Plate"].str.contains(plate)],
+            gg.aes(x="x_loc", y="y_loc"),
+        )
+        + gg.geom_point(gg.aes(fill=corr_qc_col), shape="s", size=6)
         + gg.geom_text(gg.aes(label="site_location"), color="lightgrey", size=6)
-        + gg.facet_wrap(f"~{image_cols['well']}")
+        + gg.facet_wrap("~Metadata_Well")
         + gg.coord_fixed()
         + gg.theme_bw()
-        + gg.ggtitle(f"{threshhold_compartment} Thresholds \n {plate}")
+        + gg.ggtitle(
+            f"Correlation between BC & CP DAPI images\n(Alignment quality)\n{plate}"
+        )
         + gg.theme(
             axis_text=gg.element_blank(),
             axis_title=gg.element_blank(),
             strip_background=gg.element_rect(colour="black", fill="#fdfff4"),
         )
-        + gg.scale_fill_cmap(name="Threshold")
-        + gg.labs(fill="Threshold")
+        + gg.scale_fill_cmap(name="Correlation")
     )
+
     output_file = pathlib.Path(
         output_figuresdir,
-        f"plate_layout_{threshhold_compartment}_FinalThreshold_per_well.png",
+        f"plate_layout_BC_to_CP_DAPI_correlation_per_well_{plate}.png",
     )
     if check_if_write(output_file, force, throw_warning=True):
-        compartment_finalthresh_gg.save(output_file, dpi=300, verbose=False)
+        correlation_gg.save(output_file, dpi=300, verbose=False)
+
+# Plot final compartment thresholds per well
+threshold_col_prefix = "Threshold_FinalThreshold_"
+for threshhold_compartment in ["Cells", "Nuclei"]:
```

Comment:
Not sure if I made this comment before about hardcoding these compartments - is there a way to avoid it?

Reply:
Maaaaybe...?

In a Cell Painting workflow, `Cells` and `Nuclei` are the two compartments that are always segmented. Segmenting fewer compartments is impossible because we need to identify individual cells and we must use `Nuclei` to determine `Cells`. It's possible to segment more compartments, but we don't currently do that (even in our workflow where we have many more labels) and are unlikely to do so because 1) it seems to be unnecessary and 2) it's prone to mistakes/variability and therefore requires significant hands on time.

Therefore, the thresholds we want to plot here are likely to always be `Cells` and `Nuclei`. The question is therefore should we remove hardcoding in case `Cells` and `Nuclei` compartments are ever be labeled in a different way (e.g. `cells`, `Cell`, etc.). To do that, it looks like the options are:
1) Create a new entry in options.yaml to specify the segmented compartment names. Not ideal because it's one more thing to have to enter, but it's the least prone to breaking.
2) Use `core:  compartments:` and remove `Cytoplasm` from the list because it's the only tertiary compartment (mathematically determined by subtracting one compartment from another) and any other compartment in the list will have been created by segmentation. Not ideal because then we have to find a way to account for labelling of `Cytoplasm`.
3) Use `core: cell_match_cols: cytoplasm:` and then strip the `Parent_` from the front of the compartment string. `Parent` is hardcoded into CellProfiler, so it's fine from our workflow perspective, but would mean this wouldn't work with data coming not from CellProfiler (which I thought was a goal?).

What do you think we should do?

Reply:
Thanks for laying this out - I want to capture these ideas, so I opened a new issue (#76)

But, I think there is an option 4 too, for this particular PR:

4. Leave it be for now, since it's not hurting anything at the moment.

My vote is for 4.

Reply:
I agree with leaving it alone for now!